video_creation/background.py

```diff
@@ -23,7 +23,7 @@ def download_background():
             "We need to download the Minecraft background video. This is fairly large but it's only done once. 😎"
         )
         print_substep("Downloading the background video... please be patient 🙏")
-        YouTube("https://www.youtube.com/watch?v=n_Dv4JMiwK8").streams.filter(
+        YouTube("https://www.youtube.com/watch?v=n_Dv4JMiwK8", on_progress_callback=on_progress).streams.filter(
```

Comment:
I don't think this is that good if an idea. This progress bar isn't that great and if it hangs, the user will exit the program. In my latest PR, I've submitted utils/loader.py which is super easy to use.

Reply:
> I don't think this is that good of an idea. This progress bar isn't that great and if it hangs, the user will exit the program. In my latest PR, I've submitted utils/loader.py which is super easy to use.

The progress bar here is async via callbacks tho so it can't hang.

Reply:
Makes more sense! There's two progress bars, one of which is not asynchronous and is not very good at being a progress bar.